snapcraft/extractors/_metadata.py

```diff
@@ -63,7 +73,23 @@ def get_description(self) -> str:
         """
         return self._data.get('description')
 
-    def to_dict(self) -> Dict[str, str]:
+    def get_icon(self) -> str:
+        """Return extracted icon.
+
+        :returns: Extracted icon
+        :rtype: str
+        """
+        return self._data.get('icon')
+
+    def get_desktop(self) -> List[str]:
+        """Return extracted desktop files.
+
+        :returns: Extracted desktop files ids
```

Comment:
One comment says "files", the other says "files ids". I think we should use only one term to avoid confusion. Since we rely on the fact that these are filenames, I'll vote for "desktop files".

Reply:
Yeah the term "ID" confuses me as well. I suggest we keep the language in this class as generic as possible given that multiple extractors will be using it, not just appstream.